dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java

```diff
@@ -77,8 +93,75 @@
     private static final int Y_ACCELERATION_INDEX = 1;
     private static final int Z_ACCELERATION_INDEX = 2;
 
-    private float[] gravity = { 0 , 0 , 0};
-    private final float alpha = (float) 0.8;
+    private static final int FASTEST_PUBLICATION_RATE = 200;
+
+    private ScheduledFuture publicationHandle;
+    private final Runnable publishData = new Runnable() {
+        private boolean isConnecting = false;
+
+        @Override
+        public void run() {
+            if (client.isConnected()) {
+                isConnecting = false;
+                String data = "";
+
+                // Calculate the rate between publish events
+                long currentTimeMilliseconds = System.currentTimeMillis();
+                publishRateMilliSec = currentTimeMilliseconds - lastTimeCheck;
+                lastTimeCheck = currentTimeMilliseconds;
+
+                JSONObject accelerationJson = new JSONObject();
+
+                try {
+                    accelerationJson.put(WATCH_ID_JSON_INDEX, androidId);
+
+                    accelerationJson.put(ACC_X_JSON_INDEX,
+                            xAcceleration);
+
+                    accelerationJson.put(ACC_Y_JSON_INDEX,
+                            yAcceleration);
+
+                    accelerationJson.put(ACC_Z_JSON_INDEX,
+                            zAcceleration);
+
+                    accelerationJson.put(TIMESTAMP_JSON_INDEX,
+                            currentTimeMilliseconds);
+
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+                try {
+
+                    // Convert JSON to string and publish
```

Comment:
Do we want to be doing this when the `JSONException` happens? maybe merge the two try catch